packages/aws-cdk-lib/custom-resource-config/test/custom-resource-config.test.ts

```diff
@@ -0,0 +1,547 @@
+            RetentionInDays: SET_LOG_RETENTION,
+        });
+        template.hasResourceProperties('AWS::Logs::LogGroup', {
+            RetentionInDays: DEFAULT_LOG_GROUP_RETENTION,
+        });
+        template.hasResourceProperties('AWS::Logs::LogGroup', {
+            RetentionInDays: SET_LOG_RETENTION
+        })
+    });
+    test('s3LogRetention - snapshot', () => {
+        const SET_LOG_RETENTION = logs.RetentionDays.TEN_YEARS;
+        const LOCALLY_SET_LOG_RETENTION = logs.RetentionDays.ONE_WEEK;
+
+        const app = new cdk.App();
+        CustomResourceConfig.of(app).addLogRetentionLifetime(SET_LOG_RETENTION);
+        const stack = new cdk.Stack(app);
+
+        const websiteBucket = new s3.Bucket(stack, "WebsiteBucket", {});
+        new s3deploy.BucketDeployment(stack, "s3deployLogRetention", {
+            sources: [s3deploy.Source.jsonData("file.json", { a: "b" })],
+            destinationBucket: websiteBucket,
+            logRetention: LOCALLY_SET_LOG_RETENTION
+        });
+
+        const template = Template.fromStack(stack);
+
+        expect(template.toJSON()).toMatchSnapshot();
+    });
+});
+
+describe("s3None", () => {
+    test("s3None - UnModified by Aspects", () => {
+        const app = new cdk.App();
+        const stack = new cdk.Stack(app);
+
+        const websiteBucket = new s3.Bucket(stack, "WebsiteBucket", {});
+        new s3deploy.BucketDeployment(stack, "s3deployNone", {
+            sources: [s3deploy.Source.jsonData("file.json", { a: "b" })],
+            destinationBucket: websiteBucket,
+        });
+
+        const template = Template.fromStack(stack);
+
+        template.hasResourceProperties("AWS::Lambda::Function", {
+            LoggingConfig: Match.absent(),
+        });
+        template.resourceCountIs("AWS::Logs::LogGroup", 0);
+    });
+    test("s3None - Updated LogGroup to SET_LOG_RETENTION", () => {
+        const SET_LOG_RETENTION = logs.RetentionDays.TEN_YEARS;
+
+        const app = new cdk.App();
+        CustomResourceConfig.of(app).addLogRetentionLifetime(SET_LOG_RETENTION);
+        const stack = new cdk.Stack(app);
+
+        const websiteBucket = new s3.Bucket(stack, "WebsiteBucket", {});
+        new s3deploy.BucketDeployment(stack, "s3deployNone", {
+            sources: [s3deploy.Source.jsonData("file.json", { a: "b" })],
+            destinationBucket: websiteBucket,
+        });
+
+        const template = Template.fromStack(stack);
+
+        template.hasResourceProperties("AWS::Lambda::Function", {
+            LoggingConfig: {
+                LogGroup: {
+                    Ref: Match.stringLikeRegexp("CustomCDKBucketDeployment"),
+                },
+            },
+        });
+        template.resourceCountIs("AWS::Logs::LogGroup", 1);
+        template.hasResourceProperties("AWS::Logs::LogGroup", {
+            RetentionInDays: SET_LOG_RETENTION,
+        });
+    });
+    test("s3None - assert that extra logGroup is unchanged", () => {
```

Comment:
This description is relying on context that the reader lacks; what is `s3None`,  why is there an extra log group, and what operation is supposed to leave it unchanged? 

A more informative description would be:

"CustomResourceConfig only affects Log Groups that support custom resources".